src/lib/components/files/state.ts

```diff
@@ -0,0 +1,6 @@
+export enum State {
```

Comment:
Let's move this to types.ts for now and name it LoadingState this way it can be used by other things as well. I suggest the name change so that `State` does not get confused with the application state.